### DISCUSSION TITLE
Eigenvalue analysis of a structure

### DISCUSSION THREAD

**Discussion by qc-94** [wenbinyugroup/.github#22] [VABS Related Discussions]
Hi everyone,

Regarding eigenvalue analysis, like the one proposed in iVABS examples "_anl_box_beam_eigen_", I saw that obtaining the _n_ first modes/eigenfrequencies on a structure was feasible to output. To go a step further, would it be possible to also output structure's deformed shape/vibrations modes at every vibration frequencies and have the possibility to plot displacement field (eventually normalized according to the largest displacement) at every cross sections of the structure?

Also for eigenvalue analysis, how is it possible to update stifness/compliance matrixes to take into account centrifugal forces effect on the natural frequencies of the structure, in order to be able to plot a Campbell diagram for example ?

**Comment by unitasium** [comment]
You can find the mode shapes in the output file of the eigen analysis. For this example, the file is `beam_design.dat.out`. You can use the following script to call the API to extract the data:

```python
from msgd.ext.gebt import readGEBTIn, readGEBTOut

fn_gebt_in = 'beam_design.dat'
fn_gebt_out = 'beam_design.dat.out'

beam = readGEBTIn(fn_gebt_in, 'txt')

steady_out, eigen_out = readGEBTOut(fn_gebt_out, beam)
...
```

The arrangement of the data in the output is

```rst
    Returns
    -------
    list, list
        Steady results, eigen results

        Eigen results: `[eigenvalues, eigenvectors]`

        eigenvalues = `[[eva11, eva12], [eva21, eva22], ...]`

        eigenvectors = `[eve1, eve2, ...]`

        for each eigenvector evei: `[point results, member results]`

        for each point result: `[[x11, x12, x13, u11, u12, u13, t11, t12, t13], ...]`

        for each memebr result: `[[[x11, x12, x13, u11, u12, u13, t11, t12, t13], [], ...], ...]`
```
where `x`s are beam node coordinates, `u`s are displacements, and `t`s are rotations.

For rotating beams, the centrifugal forces have been taken into consideration in GEBT already. You only need to specify a rotating speed. To make a campbell plot, you can make the rotating speed as a design variable and do a parametric study from 0 to a certain value. Then you can get different frequencies and mode shapes at different speeds.

**Reply by qc-94** [reply]
Many thanks for your answer, I created a script called "script-post.py" with what you gave to me, but it seems it didn't work out in my case for running the script after the modal analysis and after obtaining the beam_design.dat.out file :

![image](https://github.com/user-attachments/assets/623b5a28-f5e1-4110-85b1-678af54b778c)

Do you know if I am doing something wrong ? 

For GEBT analysis do you have an exemple where a rotating speed is specified ? Regarding GEBT Manual in the GIT (https://wenbinyugroup.github.io/ivabs/ref/gebt/index.html) it seems that the section is empty,  just to say that if you have any documentation, I'm more than interrested.

Thnak you for your help,
